ide/projectapi/src/org/netbeans/spi/project/NestedClass.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.spi.project;
+
+import java.util.Objects;
+import org.openide.filesystems.FileObject;
+
+/**
+ * Structure representing an identification of a nested class in a file.
+ * 
+ * <p>
+ * <code>NestedClass</code> can be used to represent nested classes within parent class
+ * Example:
+ * If we have following structure: ParentClass (parent-of) ChildClass1 (parent-of) ChildClass2,
+ * for ChildClass1 className field would contain "ChildClass1" and topLevelClassName would contain "ParentClass",
+ * for ChildClass2 className field would contain "ChildClass1.ChildClass2" and topLevelClassName would contain "ParentClass"
+ * </p>
+ * 
+ * @author Dusan Petrovic
+ * 
+ * @since 1.99
+ */
+public final class NestedClass {
```

Comment:
Not really a request to change anything, just an observation: this whole angle around `NestedClass` seems mostly orthogonal to the run tests in parallel feature, and could probably have been a separate PR.